aten/src/ATen/cuda/CUDAGenerator.cpp

```diff
@@ -44,25 +45,25 @@ CUDAGenerator* getDefaultCUDAGenerator(DeviceIndex device_index) {
     TORCH_CHECK(idx >= 0 && idx < num_gpus);
   }
   std::call_once(cuda_gens_init_flag[idx], [&] {
-    default_gens_cuda[idx] = std::make_shared<CUDAGenerator>(idx);
+    default_gens_cuda[idx] = make_generator<CUDAGenerator>(idx);
     default_gens_cuda[idx]->seed();
   });
-  return default_gens_cuda[idx].get();
+  return default_gens_cuda[idx];
 }
 
 /**
  * Utility to create a CUDAGenerator. Returns a shared_ptr
  */
-std::shared_ptr<CUDAGenerator> createCUDAGenerator(DeviceIndex device_index) {
+Generator createCUDAGenerator(DeviceIndex device_index) {
   std::call_once(num_gpu_init_flag, initCUDAGenVector);
   DeviceIndex idx = device_index;
   if (idx == -1) {
     idx = c10::cuda::current_device();
   }
   TORCH_CHECK(idx >= 0 && idx < num_gpus, "The device_index is invalid.");
-  auto gen = std::make_shared<CUDAGenerator>(idx);
-  gen->set_current_seed(default_rng_seed_val);
-  gen->set_philox_offset_per_thread(0);
+  auto gen = make_generator<CUDAGenerator>(idx);
```

Comment:
With my C++ API hat on, we should endeavor to make the API for allocating a CUDA generator look as much as possible as the Python API. (So, not `make_generator`, in general! `make_generator` still a useful helper function for internal implementation reasons though, I guess.)

Reply:
possible solution is adding `Generator(Device device)`, please see the next PR

Reply:
Constructor with device is probably not the way we should do it :) We'll talk about it on the other PR.